packages/kbn-journeys/journey/journey_ftr_harness.ts

```diff
@@ -117,8 +119,12 @@ export class JourneyFtrHarness {
   }
 
   private async onSetup() {
+    // We start browser and init page in the first place
+    await this.setupBrowserAndPage();
+    // We allow opt-in beforeSteps hook to manage Kibana/ES state
+    await this.journeyConfig.getBeforeStepsFn(this.getCtx());
+    // Loading test data
     await Promise.all([
-      this.setupBrowserAndPage(),
```

Comment:
Since playwright `page` is part of BaseStepCtx, we need to run `setupBrowserAndPage` in the first place (page init is part of `setupBrowserAndPage`)

I'm not sure if we need to expose page context in `beforeSteps` hook, but in theory somebody may want to do actions in browser as well. Having different contexts for beforeSteps and step was on my mind, but seems to be too much extra work while restriction is not clearly needed.

Thoughts?

Reply:
:+1:  - when/if there's a use case seems reasonable, but until then I don't think there's a need.

Reply:
I have not checked yet, but is the IDE tooling working with this change?